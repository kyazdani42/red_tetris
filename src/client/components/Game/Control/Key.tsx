import * as React from 'react';

import { KeyImg, KeyStyle } from './styles';

interface Props {
  emitter: () => void;
  type: keyType;
  keyPressed: keyType | null;
}

const Key: React.SFC<Props> = ({ keyPressed, emitter, type }) => (
  <KeyStyle onClick={emitter}>
    <KeyImg
      className="key-img"
      src={`/assets/icon${type === ' ' ? 'space' : type}.png`}
      isPressed={keyPressed === type}
    />
  </KeyStyle>
);

export default Key;
