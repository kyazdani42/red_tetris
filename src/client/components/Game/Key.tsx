import * as React from 'react';
import styled from 'styled-components';

interface Props {
  emitter: () => void;
  type: keyType;
  keyPressed: keyType | null;
}

const KeyStyle: any = styled.div`
  width: 40px;
  height: 40px;
  color: #000;
`;

const Img: any = styled.img`
  background-color: ${(props: any) => props.isPressed ? '#555' : '#fff'};
  filter: invert(1);
  width: 40px;
`;

const Key = ({ keyPressed, emitter, type }: Props) => (
  <KeyStyle
    onClick={emitter}
  >
    <Img src={`/assets/icon${type === ' ' ? 'space' : type}.png`} isPressed={keyPressed === type} />
  </KeyStyle>
);

export default Key;
