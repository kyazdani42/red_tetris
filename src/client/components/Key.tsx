import * as React from 'react';
import styled from 'styled-components';

import '../../../assets/icon-down.png';
import '../../../assets/icon-left.png';
import '../../../assets/icon-right.png';
import '../../../assets/icon-space.png';
import '../../../assets/icon-up.png';

interface Props {
  emitter: () => void;
  type: keyType;
  keyPressed: keyType | null;
}

const KeyStyle: any = styled.div`
  width: 10%;
  height: 10%;
  color: #000;
  background-color: ${(props: any) => props.isPressed ? '#555' : '#fff'};
`;

const Key = ({ keyPressed, emitter, type }: Props) => (
  <KeyStyle
    onClick={emitter}
    isPressed={keyPressed === type}
  >
    <img src={`/assets/icon-${type === ' ' ? 'space' : type}.png`} />
  </KeyStyle>
);

export default Key;
