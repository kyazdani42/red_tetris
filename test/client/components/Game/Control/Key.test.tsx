// tslint:disable
import * as React from 'react';
import { shallow } from 'enzyme';

import Key from '../../../../../src/client/components/Game/Control/Key';

describe('testing the component', () => {
  const props = {
    emitter: jest.fn(),
    type: ' ',
    keyPressed: null
  };
  const wrapper = shallow(<Key {...props} />);
  it('smoke tests the component', () => {
    expect(wrapper.update()).toMatchSnapshot();
  });
  it('checks that key dispatch the emitter', () => {
    wrapper.simulate('click');
    expect(props.emitter).toHaveBeenCalled();
  });
  it('verify the src with space', () => {
    const image = wrapper.find('.key-img');
    expect(image.prop('src')).toEqual('/assets/iconspace.png');
  })
  it('verify the src with test', () => {
    wrapper.setProps({ type: 'test'});
    const image = wrapper.find('.key-img');
    expect(image.prop('src')).toEqual('/assets/icontest.png');
  })
  it('checks that keyPress is false', () => {
    const image = wrapper.find('.key-img');
    expect(image.prop('isPressed')).toBeFalsy();
  });
  it('checks that keyPress is true', () => {
    wrapper.setProps({ keyPressed: ' '});
    const image = wrapper.find('.key-img');
    expect(image.prop('isPressed')).toBeTruthy()
  });
});

