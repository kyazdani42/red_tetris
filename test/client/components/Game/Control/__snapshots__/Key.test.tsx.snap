// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the component smoke tests the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Key
    emitter={[Function]}
    keyPressed={null}
    type=" "
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED
        className="key-img"
        isPressed={false}
        src="/assets/iconspace.png"
      />,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "key-img",
        "isPressed": false,
        "src": "/assets/iconspace.png",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  background-color: ",
            [Function],
            ";
  filter: invert(1);
  width: 3vmax;
  max-width: 30px;
",
          ],
        },
        "displayName": "styled.img",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "img",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  color: #000;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED
          className="key-img"
          isPressed={false}
          src="/assets/iconspace.png"
        />,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "key-img",
          "isPressed": false,
          "src": "/assets/iconspace.png",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  background-color: ",
              [Function],
              ";
  filter: invert(1);
  width: 3vmax;
  max-width: 30px;
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  color: #000;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
